### PR TITLE
Add io.github.nemuelw.protodesk

### PR DESCRIPTION
**Pull request by nemuelw**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  Unofficial desktop application for [Proton](https://proton.me/) that enables Proton users to access Proton mail,
calendar, and drive from the desktop.
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/nemuelw/protodesk

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.nemuelw.protodesk.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - /include
+      - /lib/pkgconfig
+      - /sbin
+      - /share/et
+      - /share/examples
+
+  - name: Protodesk
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/nemuelw/protodesk/73e45c4ec2b8f64c82fdd35dae90c49ba77e5024/flathub/io.github.nemuelw.protodesk.desktop
```
Comment (hfiguiere) [inline io.github.nemuelw.protodesk.yml:35]: this is never installed

File: io.github.nemuelw.protodesk.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        sha512: d9804756efcfc05e7827097f235fdd59b918ad3d390abb3f099f24c02a16c158eab3757b0b5d937ee29a0d1757f2897d78352f6fd8f26e60aaf7a4b7418efa99
+        dest: ./share/applications/io.github.nemuelw.protodesk.desktop
+
+      - type: file
+        url: https://raw.githubusercontent.com/nemuelw/protodesk/e98976440e3ea147f77f9fc46e468b9d82791701/flathub/io.github.nemuelw.protodesk.metainfo.xml
```
Comment (hfiguiere) [inline io.github.nemuelw.protodesk.yml:40]: this is never installed

File: io.github.nemuelw.protodesk.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        sha512: 7c74e36730a732da9a9fb09e3ecc7d15b8a77ae2f8acf8529d967c3dd1c74dca4fa96863a0b9a67f8991e2d8432240d43a3d7f2f761c2a66da134dc46ee709c9
+        dest: ./share/metainfo/io.github.nemuelw.protodesk.metainfo.xml
+
+      - type: file
+        url: https://raw.githubusercontent.com/nemuelw/protodesk/73e45c4ec2b8f64c82fdd35dae90c49ba77e5024/assets/logo.svg
```
Comment (hfiguiere) [inline io.github.nemuelw.protodesk.yml:45]: this is never installed

File: io.github.nemuelw.protodesk.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        sha512: 1ea4475c7673457fd1c8c8d24089fddebe9653081c3db30d93a86b2e71284470bd04ad987632214e0030c4a381700c556747aa93c867545e9cf494b259b71003
+        dest: ./share/icons/hicolor/scalable/apps/io.github.nemuelw.protodesk.svg
+
+      - type: archive
+        url: https://github.com/nemuelw/protodesk/releases/download/v1.2.6/protodesk-v1.2.6_amd64.tar.gz
```
Comment (hfiguiere) [inline io.github.nemuelw.protodesk.yml:50]: why not building from source instead?

Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml:50]: The app will be distributed via other channels besides Flathub, and so having a centralized upstream build is the better option.

Comment (hfiguiere) [reply io.github.nemuelw.protodesk.yml:50]: That's a myth. Also if it's so centralized with doesn't it include icon, desktop file and appstream file?

Also I see it does include Qt which is a bad idea. The runtime has patches in Qt to work better with Flatpak and Linux in general. Instead you bloat the distribution.

Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml:50]: As you might have seen, the flatpak specific files are in their own directory. Upon having the builds upstream, it's more straightforward to also create deb, rpm or snap.

Comment (hfiguiere) [reply io.github.nemuelw.protodesk.yml:50]: neither of these files are flatpak specific. They work for any linux distro

and they are not even installed.

Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml:50]: I was referring to the files in the [`flathub` directory](https://github.com/nemuelw/protodesk/tree/main/flathub) specifically.

Comment (hfiguiere) [reply io.github.nemuelw.protodesk.yml:50]: as I did.

Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml:50]: Gotcha! Thanks for the clarification.

Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml:50]: Anything else I might be missing here?

**Comment by nemuelw** [comment]
bot, build io.github.nemuelw.protodesk

File: io.github.nemuelw.protodesk.yml (hunk 4 above)
Comment (hfiguiere) [reply io.github.nemuelw.protodesk.yml:50]: > and they are not even installed.

as I said, but I guess it needed to be demonstrated:

```
        "appstream-metainfo-missing: No metainfo file for io.github.nemuelw.protodesk was found in /app/share/metainfo or /app/share/appdata"
```

File: io.github.nemuelw.protodesk.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        url: https://github.com/nemuelw/protodesk/releases/download/v1.2.6/protodesk-v1.2.6_arm64.tar.gz
+        sha512: 439283d977d0c919a72e7712b15500bdba38d271834b6aaad0cb6d77fe882a58fe8f0f0af6b48e13a0e2003aad2d768bbd07749caadf6df9a2cf85c24fe7f3db
+        dest: ./dist
+        only-arches:
+          - aarch64
+
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - cp -r ./dist/* /app/bin/
```
Comment (bbhtt) [inline io.github.nemuelw.protodesk.yml:66]: this should be after `name` of the module.

File: io.github.nemuelw.protodesk.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+      - type: file
+        url: https://raw.githubusercontent.com/nemuelw/protodesk/bb1349bb1a2d5ca1c88d940ca06787b0cc7baec9/flathub/io.github.nemuelw.protodesk.metainfo.xml
+        sha512: eec8d393ffd43ea119ef2e3d45c45f5a24ef46c0a9a41822a34f4bf6420b0547e779edbf03efeaabb8126705604c461653341a371007a24785cc4554b29ff65e
+        dest: ./share/metainfo/io.github.nemuelw.protodesk.metainfo.xml
+
+      - type: file
+        url: https://raw.githubusercontent.com/nemuelw/protodesk/bb1349bb1a2d5ca1c88d940ca06787b0cc7baec9/assets/logo.svg
+        sha512: 1ea4475c7673457fd1c8c8d24089fddebe9653081c3db30d93a86b2e71284470bd04ad987632214e0030c4a381700c556747aa93c867545e9cf494b259b71003
+        dest: ./share/icons/hicolor/scalable/apps/io.github.nemuelw.protodesk.svg
```
Comment (bbhtt) [inline io.github.nemuelw.protodesk.yml:47]:
```suggestion

      - type: file
        url: https://raw.githubusercontent.com/nemuelw/protodesk/bb1349bb1a2d5ca1c88d940ca06787b0cc7baec9/flathub/io.github.nemuelw.protodesk.metainfo.xml
        sha512: eec8d393ffd43ea119ef2e3d45c45f5a24ef46c0a9a41822a34f4bf6420b0547e779edbf03efeaabb8126705604c461653341a371007a24785cc4554b29ff65e

      - type: file
        url: https://raw.githubusercontent.com/nemuelw/protodesk/bb1349bb1a2d5ca1c88d940ca06787b0cc7baec9/assets/logo.svg
        sha512: 1ea4475c7673457fd1c8c8d24089fddebe9653081c3db30d93a86b2e71284470bd04ad987632214e0030c4a381700c556747aa93c867545e9cf494b259b71003
```

Comment (bbhtt) [reply io.github.nemuelw.protodesk.yml:47]: This dest doesn't do what you think it does.

**Comment by nemuelw** [comment]
bot, build io.github.nemuelw.protodesk

File: io.github.nemuelw.protodesk.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    build-commands:
+      - install -Dm644 flathub/io.github.nemuelw.protodesk.desktop /app/share/applications/io.github.nemuelw.protodesk.desktop
+      - install -Dm644 assets/logo.svg /app/share/icons/hicolor/scalable/apps/io.github.nemuelw.protodesk.svg
+      - rm /app/share/metainfo/io.qt.PySide.BaseApp.metainfo.xml
+      - install -Dm644 flathub/io.github.nemuelw.protodesk.metainfo.xml /app/share/metainfo/io.github.nemuelw.protodesk.metainfo.xml
+      - cp -r ./assets /app/bin
+      - install -Dm755 app.py /app/bin/app.py
+      - echo '#!/bin/sh' > /app/bin/protodesk
+      - echo 'python3 /app/bin/app.py' >> /app/bin/protodesk
+      - chmod +x /app/bin/protodesk
```
Comment (bbhtt) [inline io.github.nemuelw.protodesk.yml:37]: use a script source https://docs.flatpak.org/en/latest/module-sources.html#shell-and-script-sources

File: io.github.nemuelw.protodesk.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+sdk: org.kde.Sdk
+base: io.qt.PySide.BaseApp
+base-version: '6.7'
+cleanup-commands:
+  - /app/cleanup-BaseApp.sh
+command: protodesk
+
+finish-args:
+  - --device=dri
+  - --filesystem=host
```
Comment (bbhtt) [inline io.github.nemuelw.protodesk.yml:13]: why is host needed? Does it not use filechooser portals?

**Comment by bbhtt** [comment]
bot, build io.github.nemuelw.protodesk

File: io.github.nemuelw.protodesk.yml (hunk 8 above)
Comment (nemuelw) [reply io.github.nemuelw.protodesk.yml]: It sure does. In choosing to add host I hadn't considered that filechooser portals is enough for the application's needs.

**Comment by bbhtt** [comment]
The app here seems to be a single Python script. How is this different from using Proton mail on the web browser or as a PWA?

**Comment by nemuelw** [comment]
> The app here seems to be a single Python script. How is this different from using Proton mail on the web browser or as a PWA?
> 
> I'm a bit confused as to why this needs to be a separate app.

It is intended to accomodate the shortcoming that there are distinct apps, web urls and by extension PWAs for the different Proton services. Proton doesn't just offer Proton mail. There is Proton calendar and Proton drive. A single unified Proton account is used across all these services. So, why visit separate urls in the browsers, install separate PWAs or outright desktop apps for Mail, Calendar & Drive when you could have just a single app that lets you conveniently interact with all three from a single location? That is the gap Protodesk seeks to fill.

**Comment by bbhtt** [comment]
That doesn't sound very convincing me to for what effectively is toggling between a few pinned tabs in the browser or opening two different PWAs.

There is not enough functionality or features right now to accept this as an app.

> This is also just the first release. Once it is in the hands of diverse Proton users out there it'll be easier to make decisions going forward.

If more things are added in the future, this can be reconsidered.

**Pull request closed without merging**